shell/app/node_main.cc

```diff
@@ -141,14 +196,22 @@ int NodeMain(int argc, char* argv[]) {
     if (flags_exit_code != 0)
       exit(flags_exit_code);
 
+#if BUILDFLAG(IS_LINUX)
+    // Hack around with the argv pointer. Used for process.title = "blah".
+    argv = uv_setup_args(new_argc, new_argv);
+
+    std::vector<std::string> args(new_argv, new_argv + new_argc);
+#else
     // Hack around with the argv pointer. Used for process.title = "blah".
     argv = uv_setup_args(argc, argv);
 
     std::vector<std::string> args(argv, argv + argc);
+#endif
     std::unique_ptr<node::InitializationResult> result =
         node::InitializeOncePerProcess(
             args,
             {node::ProcessInitializationFlags::kNoInitializeV8,
+             node::ProcessInitializationFlags::kNoDefaultSignalHandling,
```

Comment:
🤔 should we only pass this flag if there's a crashpad handler around? i.e. if there's no crashpad handler PID/FD passed, let Node do its own default signal handling.

Also, what if anything do we lose by disabling the default signal handling?

Reply:
Would it make sense to gate this to Linux only, and only if the crashpad is initialized? Something like:

```suggestion
#if BUILDFLAG(IS_LINUX) && fd_string
             node::ProcessInitializationFlags::kNoDefaultSignalHandling,
#endif
```

Reply:
yeah, seems like it should be gated on both 👍🏻

Reply:
> Also, what if anything do we lose by disabling the default signal handling?

Looking at this in Nodejs, the flag is used to ensure the following handlers

1) Handle SIGUSR1 for inspector to start debugging a process
2) Perform custom cleanup for SIGINT, SIGTERM cases

I think 1) is not of interest for crashpad and for 2) Nodejs does a `raise()` call after handling the signal which should surface the signals that crashpad can still capture. So I think we don't need to disable the default signal handlers. @nornagon thoughts ?

Reply:
Nvm, we need the flag for crashpad case since Nodejs restores all signal handlers to default on start https://github.com/nodejs/node/blob/ed3604cd64765cb6dfb0b7160ce1b829666d3194/src/node.cc#L418 and since we do crashpad init before Nodejs init this will remove all the crash signal handlers set by the former.

Reply:
Alt: address without the flag https://github.com/electron/electron/pull/36460/commits/be1566278fb120f94bd329fe53e9b9aa98f01be5

Reply:
Tried this without moving initialization and it looks like it does cause some of the node tests/parallel to fail!: https://github.com/electron/electron/commit/40cd488d20532d0b8353400d6053546181f49cab I think moving the initialization is the right call 👍